src/robot.cpp

```diff
@@ -73,15 +74,17 @@ auto Robot::from(std::shared_ptr<moveit::core::RobotModel const> const& model,
 }
 
 auto get_link_indexes(std::shared_ptr<moveit::core::RobotModel const> const& model,
-                      std::vector<std::string> const& names) -> std::vector<size_t> {
+                      std::vector<std::string> const& names)
+    -> tl::expected<std::vector<size_t>, std::string> {
     std::vector<size_t> indexes(names.size(), 0);
-    std::transform(names.cbegin(), names.cend(), indexes.begin(), [&model](auto const& name) {
+    for (auto const& name : names) {
         auto const* link_model = model->getLinkModel(name);
         if (!link_model) {
-            throw std::invalid_argument(fmt::format("link not found: {}", name));
+            return tl::make_unexpected(fmt::format("link not found: {}", name));
         }
-        return link_model->getLinkIndex();
-    });
+        indexes.push_back(link_model->getLinkIndex());
+    }
```

Comment:
Had to replace transform with for loop because you can't return the outer function from within unary function used by the algorithm.

If you'd prefer I continue to use algorithms I could use find_if to detect if any of the link names are invalid, then do the transform on a separate call. Technically an approach like this will be more parallelizable however this function is only called during the initialization of the IK solver (normally on the order of once per execution of the process so I would argue against making this portion of the code more complex than it needs to be).

Reply:
I don't think we need to bend over backwards for such a simple loop. A `for` loop seems like a perfectly fine solution.